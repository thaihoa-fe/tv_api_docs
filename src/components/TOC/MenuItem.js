import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { MdBookmarkBorder } from 'react-icons/md';
import { Link as GatsbyLink } from 'gatsby';

const topLinkStyles = css`
  margin-top: 12px;
  span {
    text-transform: uppercase;
    font-size: 16px;
    font-weight: 600;
  }
`;

const Link = styled(GatsbyLink)`
  display: block;
  height: 38px;
  line-height: 38px;
  width: 100%;
  align-items: center;
  color: ${props => (props.selected ? '#0099e5' : '#4c555a')};
  background: ${props => (props.selected ? 'rgba(219, 239, 249, 0.6)' : 'transparent')};
  font-weight: ${props => (props.selected ? '600' : '500')};
  text-decoration: none;
  padding: ${props => `0 8px 0 ${props.level * 15}px`};
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  ${props => (props.level === 1 ? topLinkStyles : null)}
  span {
    height: 28px;
    line-height: 28px;
    text-align: left;
  }
  svg {
    display: ${props => (props.level === 3 ? 'inline' : 'none')};
    margin-right: 8px;
    vertical-align: middle;
  }
  strong {
    font-size: 14px;
    display: ${props => (props.level === 2 ? 'inline' : 'none')};
    margin-right: 8px;
  }
  &:hover {
    color: #0099e5;
    background: rgba(219, 239, 249, 0.6);
  }
`;

function MenuItem({ index, level, onClick, text, path, selected }) {
  return (
    <Link
      level={level}
      to={path}
      title={text}
      selected={selected}
    >
      <MdBookmarkBorder />
      {index > 0 && <strong>{`${index}. `}</strong>}
      <span>{text}</span>
    </Link>
  );
}

MenuItem.propTypes = {
  index: PropTypes.number,
  onClick: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  level: PropTypes.number.isRequired,
  selected: PropTypes.bool,
};

MenuItem.defaultProps = {
  selected: false,
  index: null,
};

export default MenuItem;
