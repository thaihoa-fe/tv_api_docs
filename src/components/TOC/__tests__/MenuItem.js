import React from 'react';
import renderer from 'react-test-renderer';

import MenuItem from '../MenuItem';

describe('MenuItem', () => {
  it('renders level 1 correctly', () => {
    const tree = renderer
      .create(<MenuItem onClick={jest.fn()} text="Introduction" path="/introduction" level={1} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders level 2 correctly', () => {
    const tree = renderer
      .create(
        <MenuItem
          onClick={jest.fn()}
          text="Introduction"
          path="/introduction"
          level={2}
          id="introduction"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
