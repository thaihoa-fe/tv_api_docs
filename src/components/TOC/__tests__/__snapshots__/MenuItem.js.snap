// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem renders level 1 correctly 1`] = `
<a
  className="sc-bdVaJa gDraOD"
  href="/introduction"
  level={1}
  selected={false}
  title="Introduction"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
    />
  </svg>
  <span>
    Introduction
  </span>
</a>
`;

exports[`MenuItem renders level 2 correctly 1`] = `
<a
  className="sc-bdVaJa giLJLt"
  href="/introduction"
  level={2}
  selected={false}
  title="Introduction"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
    />
  </svg>
  <span>
    Introduction
  </span>
</a>
`;
